components/brave_wallet/common/hash_utils.cc

```diff
@@ -75,4 +79,14 @@ Ripemd160HashArray Hash160(base::span<const uint8_t> input) {
   return result;
 }
 
+SHA512HashArray HmacSha512(base::span<const uint8_t> key,
+                           base::span<const uint8_t> data) {
+  SHA512HashArray hmac;
+  unsigned int out_len = 0;
+  CHECK(HMAC(EVP_sha512(), key.data(), key.size(), data.data(), data.size(),
```

Comment:
I don't know if we want to crash the browser if this fails, can it fail for any reason other than running out of memory?

Reply:
That's similarly CHECK'ed in many places like this
https://source.chromium.org/chromium/chromium/src/+/main:remoting/protocol/webrtc_transport.cc;l=829-831;drc=b31f44704280a3ec2acb6d7c0542cbe0ce47e53f;bpv=1;bpt=1

Reply:
Nevertheless now using `crypto::hmac::SignSha512`  novelty